pkg/cloudprovider/providers/azure/azure_instances.go

```diff
@@ -29,6 +29,16 @@ import (
 
 // NodeAddresses returns the addresses of the specified instance.
 func (az *Cloud) NodeAddresses(name types.NodeName) ([]v1.NodeAddress, error) {
+	if az.UseInstanceMetadata {
+		text, err := QueryMetadataText("instance/network/interface/0/ipv4/ipAddress/0/privateIpAddress")
```

Comment:
Where possible was meant to mean: "where the metadata service provides the information" as opposed to "fallback if there are errors"

We don't fallback in other places where we use other metadata services:

https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/aws.go#L779

So I think we probably don't want to here, either.